File: sources/Android/android-communications/library/src/test/java/com/polar/sdk/api/model/PolarDeviceUuidTest.kt
package com.polar.sdk.api.model

import junit.framework.TestCase.assertEquals
import org.junit.Test

class PolarDeviceUuidTest {

    @Test
    fun `fromDeviceId() should generate Polar device UUID String`() {
        // Arrange
        val deviceId = "89643A20"
        val expectedUuid = "0e030000-0084-0000-0000-000089643A20"

        // Act
        val generatedUuid = PolarDeviceUuid.fromDeviceId(deviceId)

        // Assert
        assertEquals(expectedUuid, generatedUuid)
    }

    @Test(expected = IllegalArgumentException::class)
    fun `fromDeviceId() should throw exception if device ID is invalid`() {
        // Arrange
        val invalidDeviceId = "123456789"

        // Act & Assert
        PolarDeviceUuid.fromDeviceId(invalidDeviceId)
    }
}

File: sources/iOS/ios-communications/Tests/PolarBleSdkTests/PolarDeviceUuidTest.swift
//  Copyright © 2024 Polar. All rights reserved.

import Foundation
import XCTest

class PolarDeviceUuidTest: XCTestCase {

    func testFromDeviceIdShouldGeneratePolarDeviceUuidString() throws {
        // Arrange
        let deviceId = "89643A20"
        let expectedUuid = "0e030000-0084-0000-0000-000089643A20"

        // Act
        let generatedUuid = try PolarDeviceUuid.fromDeviceId(deviceId)

        // Assert
        XCTAssertEqual(expectedUuid, generatedUuid)
    }

    func testFromDeviceIdShouldThrowExceptionIfDeviceIdIsInvalid() {
        // Arrange
        let invalidDeviceId = "123456789"

        // Act & Assert
        XCTAssertThrowsError(try PolarDeviceUuid.fromDeviceId(invalidDeviceId)) { error in
            XCTAssertTrue(error is PolarDeviceUuid.PolarDeviceUuidError)
            if case let PolarDeviceUuid.PolarDeviceUuidError.invalidDeviceIdLength(expected, actual) = error {
                XCTAssertEqual(expected, 8)
                XCTAssertEqual(actual, 9)
            } else {
                XCTFail("Expected invalidDeviceIdLength error")
            }
        }
    }
}
